src/typing/type_inference_js.ml

```diff
@@ -3065,6 +3065,32 @@ and mk_proptype cx = Ast.Expression.(function
           (_, {Ast.Identifier.name = "oneOf"; _ });
          _
       };
+      arguments = [Expression (_, Array { Array.
+        elements = es
+      })];
+    } ->
+      let rec mk_one_of es t = match (es, t) with
+        | Some (Expression (loc, Literal {
+            Ast.Literal.value = Ast.Literal.String value
+          })) :: tl,
+          UnionT (r, ts) ->
+            let reason = mk_reason "string literal type" loc in
+            let string_literal = EnumT
+              (reason,
+               Flow_js.mk_object_with_map_proto cx reason
+                 (SMap.singleton value AnyT.t) (MixedT reason)) in
+            mk_one_of tl (UnionT (r, string_literal :: ts))
+        | [], _ -> t
+        | _  -> AnyT.at vloc in
+      let reason = mk_reason "oneOf" vloc in
+      mk_one_of es (UnionT (reason, []))
```

Comment:
Seems like your comment disappeared. I had composed this:

Err, actually, the wildcard is to bail, not for non-UnionT arguments, but for non-literal strings.

I could have the recursive function return a `t list option`, to indicate failure, but then I'd just be un-re-wrapping the option.

I could separate the "is everything a string literal" check from the "let's create a type" part, but I'm not sure that would be more clear.

I'm still learning OCaml, so if there's another preferable way way (more procedural?), I'm happy to change this.